browser/ui/tabs/brave_tab_strip_model.cc

```diff
@@ -96,3 +111,21 @@ void BraveTabStripModel::CloseTabs(base::span<int> indices,
   }
   TabStripModel::CloseTabs(contentses, close_types);
 }
+
+void BraveTabStripModel::SetCustomTitleForTab(int index,
+                                              const std::u16string& title) {
+  CHECK(base::FeatureList::IsEnabled(tabs::features::kBraveRenamingTabs));
```

Comment:
I feel we have to many of these `CHECK`s everywhere, and this doesn't seem to be something upstream does for these feature flags. I feel we need to properly understand the flow of these calls, because these `CHECK`s are not free, and they are also noisy.

Reply:
Do you mean that we can skipping putting CHECK(feature) everywhere by having a few CHECK() at proper entry points?

Reply:
Actually, I've been putting these features checks after getting code reviews saying that we should put feature checks for all functions/methods that shouldn't be run without flag, so that we can guarantee that code behind flags are not involved when flag is off. Do you think it's okay to remove CHECK(feature) in this PR as we shows the context menu only when the feature flag is on?

Reply:
If you remove these `CHECK` and clarify the entry points and where the conditionals are actually occurring, then it won't hurt to add some comments in these functions explaining "Only called when X happens".

Reply:
Removed a few places where CHECK(feature) looks redundant.